codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoderTest.java

```diff
@@ -99,6 +99,9 @@
     private ChannelPromise promise;
 
     @Mock
+    private ChannelPromise voidPromise;
```

Comment:
Should we use a real voidPromise, or simulate its behavior which doesn't allow listeners to be added?  Just so the tests will validate our code's behavior with a "real" void promise?